app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt

```diff
@@ -257,8 +257,10 @@ class SearchFragment : Fragment(), BackHandler {
             REQUEST_CODE_CAMERA_PERMISSIONS -> qrFeature.withFeature {
                 it.onPermissionsResult(permissions, grantResults)
 
-                if (context!!.isPermissionGranted(Manifest.permission.CAMERA)) {
-                    permissionDidUpdate = true
+                context?.let { context: Context ->
+                    if (context.isPermissionGranted(Manifest.permission.CAMERA)) {
```

Comment:
One could also use a safe call and compare to see if it equals true. It's a bit more terse.